pkg/controller/constraint/constraint_controller.go

```diff
@@ -298,11 +308,22 @@ func logRemoval(l logr.Logger, constraint *unstructured.Unstructured, enforcemen
 }
 
 func (r *ReconcileConstraint) cacheConstraint(instance *unstructured.Unstructured) error {
+	t := r.tracker.For(instance.GroupVersionKind())
+
 	obj := instance.DeepCopy()
 	// Remove the status field since we do not need it for OPA
 	unstructured.RemoveNestedField(obj.Object, "status")
 	_, err := r.opa.AddConstraint(context.Background(), obj)
-	return err
+	if err != nil {
```

Comment:
Are there some errors that should not lead to expectations being canceled? The top candidate in my mind is when there is a failure due to a missing template b/c the template has not yet been added to OPA.

That being said, I think the way we have the timing, the template needs to be added before the watch starts. I'm unsure if relying on that internal in a separate module is too brittle.

Reply:
As discussed, let's consider adding configuration knobs to tune this behavior as a follow-up.